apps/client/src/app/(auth)/register/page.tsx

```diff
@@ -1,39 +1,50 @@
 "use client";
 
 import { EnvelopeIcon, KeyIcon, UserIcon } from "@heroicons/react/16/solid";
+import { zodResolver } from "@hookform/resolvers/zod";
 import { Button, TextInput } from "@spin-spot/components";
-import { useAuth, useCreateUser, useUsers } from "@spin-spot/services";
+import {
+  TSignUpWithCredentialsInputDefinition,
+  signUpWithCredentialsInputDefinition,
+} from "@spin-spot/models";
+import { useCreateUser } from "@spin-spot/services";
 import { useRouter } from "next/navigation";
-import { useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
 
 export default function Register() {
   const router = useRouter();
-  const auth = useAuth();
-  const users = useUsers();
   const createUser = useCreateUser();
 
-  const [email, setEmail] = useState("");
-  const [firstName, setFirstName] = useState("");
-  const [password, setPassword] = useState("");
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    trigger,
+  } = useForm<TSignUpWithCredentialsInputDefinition>({
+    resolver: zodResolver(signUpWithCredentialsInputDefinition),
+    shouldFocusError: false,
+  });
```

Comment:
Puedes cambiar el "trigger" manual por un parámetro "mode" que automaticamente hace el trigger en el evento que le digas. Por ejemplo mode:"onBlur" automaticamente aplica el onBlur={() => trigger()} para todos los campos

```suggestion
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<TSignUpWithCredentialsInputDefinition>({
    resolver: zodResolver(signUpWithCredentialsInputDefinition),
    shouldFocusError: false,
    mode: "onBlur"
  });
```